Sources/CoreGraphicsGraphics/context/CoreGraphicsContext.swift

```diff
@@ -192,6 +193,13 @@ public final class CoreGraphicsContext: GraphicsContext {
         cgContext.textPosition = CGPoint(x: bounds.minX + offset.x, y: bounds.minY + offset.y)
         CTLineDraw(line, cgContext)
     }
+
+    public func withUnsafeMutableBytes(_ body: (UnsafeMutableBufferPointer<UInt8>) throws -> Void) throws {
+        guard let dataPointer = dataPointer else {
+            throw GraphicsContextError.noRawBuffer
+        }
+        try body(dataPointer)
+    }
```

Comment:
I like the idea of direct access, but a reason I held off adding it to the graphics context is since I feel that's actually a responsibility of `BufferedImage` and would best fit into that protocol. We'd have to reconcile the architectural differences between Cairo and Core Graphics in some way though. Adding it to `CairoImage` should be fairly straightforward, since Cairo already uses a shared mutable surface, the `CGImage` type, however, is read-only (also one of the reasons why we have the `Image`/`BufferedImage` distinction at all).

Perhaps we could add a new class that wraps an owned data pointer, deals with its deallocation, that implements the `BufferedImage` protocol (since this uses plain Swift, it could probably even live in the `Graphics` target) and have `CoreGraphicsGraphics` take an instance of this class (`CoreGraphicsContext.init(width:height:format:)` could become a convenience initializer).

I'm not sure what a good name for this class would be though. Perhaps `ImageBuffer`? Or `BufferedImage` and renaming the protocol to `BufferedImageProtocol`?  Should we rename `Image` to `ImageProtocol` for consistency then?

Reply:
Interestingly such a class would also probably help implementing a pure Swift backend (see #11)